lib/consumer.ex

```diff
@@ -166,10 +166,12 @@ defmodule ExRabbitPool.Consumer do
       # process and monitors it handle crashes and reconnections
       defp handle_channel_checkout(
              {:ok, %{pid: channel_pid} = channel},
-             %{config: config, queue: queue, adapter: adapter, config: config} = state
+             %{config: config, queue: queue, adapter: adapter} = state
            ) do
         config = Keyword.get(config, :options, [])
 
+        :ok = adapter.qos(channel, queue, config)
```

Comment:
we should not call `qos` directly as the config is the config for the rabbitmq connection but not for the channel; we should create a `setup_channel` callback in the consumer, check https://github.com/esl/ex_rabbit_pool/blob/master/lib/consumer.ex#L42

then this should be converted into a `with` clause as it can fail, so maybe converting the case into a `with` and adding the `setup_channel` as part of the pipeline

```elixir
with :ok <- setup_channel(channel),
     {:ok, tag} <- adapter.consume(...) do
  ... # continue
else
  {:error, error} -> # retry logic
end
```